psi4/src/psi4/occ/manager.cc

```diff
@@ -693,6 +699,210 @@ void OCCWave::cepa_manager() {
+void OCCWave::remp_manager() {
+    timer_on("trans_ints");
+    if (reference_ == "RESTRICTED")
+        trans_ints_rhf();
+    else if (reference_ == "UNRESTRICTED")
+        trans_ints_uhf();
+    timer_off("trans_ints");
+    Eref = Escf;
+    timer_on("T2(1)");
+    set_t2_amplitudes_mp2();
+    timer_off("T2(1)");
+    timer_on("MP2 Energy");
+    mp2_energy();
+    timer_off("MP2 Energy");
+    Ecepa = Emp2;
+    Ecepa_old = Emp2;
+
+    mp2_postprocessing();
+
+    outfile->Printf("\tUsing A=%9.5g as REMP mixing parameter\n",remp_A);
+
+    // Perform REMP iterations
+    remp_iterations();
+
+    outfile->Printf("\n");
+    outfile->Printf("\t============================================================================== \n");
+    outfile->Printf("\t================ REMP2 FINAL RESULTS ================================ \n");
+    outfile->Printf("\t============================================================================== \n");
+    outfile->Printf("\tNuclear Repulsion Energy (a.u.)    : %20.14f\n", Enuc);
+    outfile->Printf("\tSCF Energy (a.u.)                  : %20.14f\n", Escf);
+    outfile->Printf("\tREF Energy (a.u.)                  : %20.14f\n", Eref);
+    outfile->Printf("\tAlpha-Alpha Contribution (a.u.)    : %20.14f\n", EcepaAA);
+    outfile->Printf("\tAlpha-Beta Contribution (a.u.)     : %20.14f\n", EcepaAB);
+    outfile->Printf("\tBeta-Beta Contribution (a.u.)      : %20.14f\n", EcepaBB);
+    outfile->Printf("\tREMP Correlation Energy (a.u.)     : %20.14f\n", Ecorr);
+    outfile->Printf("\tREMP Total Energy (a.u.)           : %20.14f\n", Ecepa);
+    outfile->Printf("\t============================================================================== \n");
+    outfile->Printf("\n");
+
+    // Set the global variables with the energies
+    variables_["REMP2 TOTAL ENERGY"] = Ecepa;
+    variables_["REMP2 CORRELATION ENERGY"] = Ecorr;
+    variables_["REMP2 SAME-SPIN CORRELATION ENERGY"] = EcepaAA + EcepaBB;
+    variables_["REMP2 OPPOSITE-SPIN CORRELATION ENERGY"] = EcepaAB;
+    variables_["REMP2 SINGLES ENERGY"] = 0.0;  // CEPA is RHF/UHF only
+    variables_["REMP2 DOUBLES ENERGY"] = EcepaAB + EcepaAA + EcepaBB;
+    variables_["CURRENT REFERENCE ENERGY"] = Eref;
+    variables_["CUSTOM SCS-REMP2 CORRELATION ENERGY"] = os_scale * EcepaAB + ss_scale * (EcepaAA + EcepaBB);
+    variables_["CUSTOM SCS-REMP2 TOTAL ENERGY"] = Escf + variables_["CUSTOM SCS-REMP2 CORRELATION ENERGY"];
+    // EcepaL = Ecepa;
+
+    std::map<std::string, double> spin_scale_energies = {{"CUSTOM", variables_["CUSTOM SCS-REMP2 TOTAL ENERGY"]},
+                                                         {"NONE", Ecepa}};
+    variables_["CURRENT ENERGY"] = spin_scale_energies[spin_scale_type_];
+    variables_["CURRENT CORRELATION ENERGY"] = variables_["CURRENT ENERGY"] - variables_["CURRENT REFERENCE ENERGY"];
+    energy_ = variables_["CURRENT ENERGY"];
+
+    // Compute Analytic Gradients
+    // NO ANALYTIC GRADIENTS YET
```

Comment:
So we can leave a comment for future work, what's missing to compute analytic gradients? I'd imagine the big issue is the lambda equations and the assembly of the response density matrices from those. The orbital Z-vector terms would need to be coded, but I'd think those are straightforward. We already have the general machinery to compute those, once we have the orbital gradient. I'd expect that to be straightforward once you have the response density matrices.

Reply:
This is completely correct, but given that the canonical methods have only mediocre performance (we assessed that recently), i am not sure whether it is worth the effort at all.

Reply:
I completely agree about efficiency. Still, I'd prefer a comment explaining this in the code, so that if somebody is looking over this five years from now, they don't have to ask, "Is this code commented because it was mostly coded but just needed some debugging, or for some other reason?"